coalib/settings/ConfigurationGathering.py

```diff
@@ -385,6 +385,20 @@ def gather_configuration(acquire_settings,
     local_bears, global_bears = fill_settings(sections,
                                               acquire_settings,
                                               log_printer)
+
+    exclude = False
+    for x in arg_list:
```

Comment:
I think it's better to use the `arg_parser` instance to do the parsing here instead of "manually" iterate through `arg_list` like this. There are existing example for how to do this https://github.com/coala/coala/blob/master/coalib/parsing/CliParsing.py#L52 or you can refer to their official docs.